pkg/ipam/node_manager.go

```diff
@@ -347,7 +356,13 @@ func (n *NodeManager) Delete(nodeName string) {
 		}
 	}
 
-	delete(n.nodes, nodeName)
+	// Delete instance from instanceManager in case Node.recalculate() use stale data
+	// when nodes from being deleted and then rejoined into the cluster
```

Comment:
Can you elaborate on that comment? I don't see how removing the node from the instances list affects `n.recalcuate()`. Can you point me to the code which will behave differently? Thanks!

Reply:
Such as I want downgrade EC2 instance configuration，so remove node from cluster and change instance type, then rejoined in the cluster. The number of instance bound ENIs may exceed the limit, it need unbound ENI. The instance of enis is changed. If the node rejoins the cluster before next `instanceAPIResync`,  `ResyncInterfacesAndIPs` return incorrect result.

Reply:
Ah, so this is relying on the code added by this PR (which forces an immediate resync if the instance is not in the list). Makes sense. 

Let's word it a little bit differently though, because this wasn't clear to me just from the comment:

```suggestion
	// Delete the instance from instanceManager. This will cause Update() to
	// invoke instancesAPIResync if this instance rejoins the cluster.
	// This ensures that Node.recalculate() does not use stale data for
	// instances which rejoin the cluster after their EC2 configuration has changed.
```